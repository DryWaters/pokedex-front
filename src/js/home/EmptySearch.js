import React from 'react';
import { Container } from 'reactstrap';

import '../../styles/home/EmptySearch.css';

class EmptySearch extends React.Component {
  render () {
    return (
      <Container className="empty-search-container">
        <h4>No Pokémon Matched Your Search!</h4>
        <p>Try these suggestions to find a Pokémon</p>
        <ul>
            <li>Reduce the number of search parameters</li>
            <li>Search for only one or two Pokémon types at a time</li>
            <li>Ensure number search is within the range</li>
        </ul>
      </Container>
    );
  }
}

export default EmptySearch;
