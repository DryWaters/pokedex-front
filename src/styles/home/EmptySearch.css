.empty-search-container {
    border: 2px solid red;
    text-align: left;
    padding: 20px;
    border-radius: 20px;
    margin-bottom: 40px;
}