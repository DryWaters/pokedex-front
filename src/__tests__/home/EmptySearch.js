import React from 'react';
import { shallow } from 'enzyme';

import EmptySearch from '../../js/home/EmptySearch';

describe('EmptySearch Component', () => {
  it('should render EmptySearch component without crashing', () => {
    shallow(<EmptySearch />);
  });

  it('should match EmptySearch component snapshot', () => {
    const wrapper = shallow(<EmptySearch />);
    expect(wrapper).toMatchSnapshot();
  });
});
