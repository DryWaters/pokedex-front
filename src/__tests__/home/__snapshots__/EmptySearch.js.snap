// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptySearch Component should match EmptySearch component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptySearch />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h4>
          No Pokemon Matched Your Search!
</h4>,
        <p>
          Try these suggestions to find a Pokemon
</p>,
        <ul>
          <li>
                    Reduce the number of search parameters
          </li>
          <li>
                    Search for only one or two Pokemon types at a time
          </li>
          <li>
                    Ensure Number search is within the range
          </li>
</ul>,
      ],
      "className": "empty-search-container",
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No Pokemon Matched Your Search!",
        },
        "ref": null,
        "rendered": "No Pokemon Matched Your Search!",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Try these suggestions to find a Pokemon",
        },
        "ref": null,
        "rendered": "Try these suggestions to find a Pokemon",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              Reduce the number of search parameters
</li>,
            <li>
              Search for only one or two Pokemon types at a time
</li>,
            <li>
              Ensure Number search is within the range
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reduce the number of search parameters",
            },
            "ref": null,
            "rendered": "Reduce the number of search parameters",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search for only one or two Pokemon types at a time",
            },
            "ref": null,
            "rendered": "Search for only one or two Pokemon types at a time",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ensure Number search is within the range",
            },
            "ref": null,
            "rendered": "Ensure Number search is within the range",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h4>
            No Pokemon Matched Your Search!
</h4>,
          <p>
            Try these suggestions to find a Pokemon
</p>,
          <ul>
            <li>
                        Reduce the number of search parameters
            </li>
            <li>
                        Search for only one or two Pokemon types at a time
            </li>
            <li>
                        Ensure Number search is within the range
            </li>
</ul>,
        ],
        "className": "empty-search-container",
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No Pokemon Matched Your Search!",
          },
          "ref": null,
          "rendered": "No Pokemon Matched Your Search!",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Try these suggestions to find a Pokemon",
          },
          "ref": null,
          "rendered": "Try these suggestions to find a Pokemon",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Reduce the number of search parameters
</li>,
              <li>
                Search for only one or two Pokemon types at a time
</li>,
              <li>
                Ensure Number search is within the range
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reduce the number of search parameters",
              },
              "ref": null,
              "rendered": "Reduce the number of search parameters",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search for only one or two Pokemon types at a time",
              },
              "ref": null,
              "rendered": "Search for only one or two Pokemon types at a time",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ensure Number search is within the range",
              },
              "ref": null,
              "rendered": "Ensure Number search is within the range",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
